parsing/HACKING.adoc

```diff
@@ -0,0 +1,9 @@
+link:parsing/parsetree.mli[Parsetree] and link:parsing/asttypes.mli[Asttypes]::
+Parsetree is an AST of the surface language of OCaml. It is well
```

Comment:
I think Drup's intent was to separate "surface language" from other representations that have been simplified (for example syntactic sugar was expanded away), eg. the typedtree or the lambda-code, which represent distinct intermediate languages. I'll reformulate as "Parsetree is an Abstract Syntax Tree (AST) representation of OCaml source code".